daemon/jobs/valuesync.go

```diff
@@ -72,29 +73,58 @@ func SyncTransactionValue() (int64, error) {
 
 	transactionTbl := model.TableNames.Transaction
 	transactionAddressTbl := model.TableNames.TransactionAddress
+	blockTbl := model.TableNames.Block
 	transactionValue := transactionTbl + "." + model.TransactionColumns.Value
+	transactionBlockHashID := transactionTbl + "." + model.TransactionColumns.BlockHashID
 	transactionID := transactionTbl + "." + model.TransactionColumns.ID
 	taCreditAmount := model.TransactionAddressColumns.CreditAmount
-	taDebitAmount := model.TransactionAddressColumns.DebitAmount
 	taTransactionID := model.TransactionAddressColumns.TransactionID
+	blockHash := blockTbl + "." + model.BlockColumns.Hash
+	blockHeight := blockTbl + "." + model.BlockColumns.Height
 
-	result, err := boil.GetDB().Exec(`
-		UPDATE ` + transactionTbl + `
-		SET ` + transactionValue + ` = (
-				SELECT COALESCE( SUM( ta.` + taCreditAmount + ` - ta.` + taDebitAmount + ` ),0.0) 
-				FROM ` + transactionAddressTbl + ` ta 
-				WHERE ta.` + taTransactionID + ` = ` + transactionID + `)`)
+	query := `
+		UPDATE ` + transactionTbl + ` 
+		INNER JOIN ` + blockTbl + ` ON ` + blockHash + ` = ` + transactionBlockHashID + `
+		SET ` + transactionValue + ` =  (
+			SELECT COALESCE( SUM( ta.` + taCreditAmount + ` ),0.0) 
+			FROM ` + transactionAddressTbl + ` ta
+			WHERE ta.` + taTransactionID + ` = ` + transactionID + ` )
+		WHERE ` + blockHeight + ` BETWEEN ? AND ?`
+
+	var from int
+	var to int
+	var affected int64
+
+	latestBlock, err := model.Blocks(qm.Select(model.BlockColumns.Height), qm.OrderBy(model.BlockColumns.Height+" DESC")).OneG()
 	if err != nil {
 		return 0, errors.Prefix(syncTransactionValues, err)
 	}
-	rowsAffected, err := result.RowsAffected()
-	if err != nil {
-		return 0, errors.Prefix(syncTransactionValues, err)
+	if latestBlock.Height == 0 {
+		return 0, errors.Prefix(syncTransactionValues, errors.Err("latest height = 0 "))
 	}
-	if rowsAffected > 0 {
-		logrus.Warn(syncTransactionValues+" rows affected ( ", rowsAffected, " )")
+	latestHeight := int(latestBlock.Height)
+	updateIncrement := 5000
+	for i := 0; i < latestHeight/updateIncrement; i++ {
```

Comment:
it just means don't run this until block height is great than 5,000 right?

Reply:
yes. but that also means that the first 5000 blocks aren't processed. Am I wrong?

Reply:
its `latestHeight` which is the highest current block  divided by `updateIncrement`. So the only way this returns 0 if is the `latestHeight` is <= 5000. Yes, this will never execute until the block height is above 5000, but we will process that in less than a minute and the very next run of this cron job it will then be greater than 5000. Does that make sense?

Reply:
Having looked more into this I see what you mean. Sure it works then.